e2e_playwright/multipage_apps_v2/mpa_v2_basics_test.py

```diff
@@ -359,3 +359,30 @@ def test_widgets_maintain_state_in_fragment(app: Page):
     wait_for_app_run(app)
 
     expect(input).to_have_value("Hello")
+
+
+def test_widget_state_reset_on_page_switch(app: Page):
+    # Regression test for GH issue 7338 for MPAv2
+
+    slider = app.locator('.stSlider [role="slider"]')
+    slider.click()
+    slider.press("ArrowRight")
+    wait_for_app_run(app, wait_delay=500)
+    expect(app.get_by_text("x is 1")).to_be_attached()
+
+    # Switch to the slow page
```

Comment:
nit: if `page 6` is a slow page / app, maybe we could add it to the variable name (e.g. `page_6_slowapp`) and file name (`page_6_(its_a_slow_app).py`). to make the expectation for that page crystal clear and avoid that someone in the future accidentally changes it content or so (also easier to grasp whats going on when reading the test)

Reply:
Will do...was trying to avoid screenshot changes, but you're right.